src/Config/ConfigPanel.ts

```diff
@@ -0,0 +1,143 @@
+import * as vscode from 'vscode';
+import {getNonce} from './GetNonce';
+
+
+export class ConfigPanel {
+  /**
+   * Track the current panel. Only allow a single panel to exist at a time.
+   */
+  public static currentPanel: ConfigPanel|undefined;
+
+  public static readonly viewType = 'one-vscode';
+
+  private readonly _panel: vscode.WebviewPanel;
+  private readonly _extensionUri: vscode.Uri;
+  private _disposables: vscode.Disposable[] = [];
+
+  public static createOrShow(extensionUri: vscode.Uri) {
+    const column =
+        vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+
+    // If we already have a panel, show it.
+    if (ConfigPanel.currentPanel) {
+      ConfigPanel.currentPanel._panel.reveal(column);
+      ConfigPanel.currentPanel._update();
+      return;
+    }
+
+    // Otherwise, create a new panel.
+    const panel = vscode.window.createWebviewPanel(
+        ConfigPanel.viewType, 'ConfigurationSettings', column || vscode.ViewColumn.One, {
+          // Enable javascript in the webview
+          enableScripts: true,
+
+          localResourceRoots: [],
+        });
+
+    ConfigPanel.currentPanel = new ConfigPanel(panel, extensionUri);
+  }
+
+  public static kill() {
+    ConfigPanel.currentPanel ?.dispose();
+    ConfigPanel.currentPanel = undefined;
+  }
+
+  public static revive(panel: vscode.WebviewPanel, extensionUri: vscode.Uri) {
+    ConfigPanel.currentPanel = new ConfigPanel(panel, extensionUri);
+  }
+
+  private constructor(panel: vscode.WebviewPanel, extensionUri: vscode.Uri) {
+    this._panel = panel;
+    this._extensionUri = extensionUri;
+
+    // Set the webview's initial html content
+    this._update();
+
+    this._panel.onDidDispose(() => this.dispose(), null, this._disposables);
+  }
+
+  public dispose() {
+    ConfigPanel.currentPanel = undefined;
+
+    // Clean up our resources
+    this._panel.dispose();
+
+    while (this._disposables.length) {
+      const x = this._disposables.pop();
+      if (x) {
+        x.dispose();
+      }
+    }
+  }
+
+  private async _update() {
+    const webview = this._panel.webview;
+    webview.html = this._getHtmlForWebview(webview);
+  }
+
+  private _getHtmlForWebview(webview: vscode.Webview) {
+    // Use a nonce to only allow specific scripts to be run
+    const nonce = getNonce();
+
+    return `<!DOCTYPE html>
```

Comment:
How about use a html file, read it and return the content?

Reply:
i think that's better for review, but vscode extension has it's own file path system. so i worry about if i separate html section, i can't import css, js on html.

Reply:
uri 
```typescript
 const stylesResetUri = webview.asWebviewUri(
        vscode.Uri.joinPath(this._extensionUri, 'media/configuration-settings', 'reset.css'));
```
and import
```javascript
`<link href="${stylesResetUri}" rel="stylesheet">`
```
like this

Reply:
Sorry, I don't have much knowledge on this topic... 
I don't see `reset.css` in this `html` looking string.
What did I miss?

Reply:
i don't understand exact mean of your question.
if your question is "there is no such reset.css in this commit, where is reset.css coming from?", then reset.css is in pr #111 and isn't merged yet. so i didn't import css on this pull request.
if your question is about example i provide, styleResetUri means path to reset.css when vscode extension is  executed.
i don't know it is enough answer.

Reply:
maybe this document helps
https://code.visualstudio.com/api/extension-guides/webview#controlling-access-to-local-resources

Reply:
Can https://stackoverflow.com/questions/51236058/use-html-files-in-visual-studio-code-extension-development help?

Reply:
Yes it's helpful. but still import problem is remained. 
In this pull request, it's just html but we will try to import css and javascript files for webview later.
I still don't know how to import css and javasciprt on html in this way

Reply:
OK, (1) let's move `return `<!DOCTYPE html>` to a function, (2) in that function let's return empty `''` string and leave a `TODO  return proper HTML` comment.
And the when you get to know how it should work, we can revise them.

Reply:
This is because I don't want to put HTML source in the typescript code.

Reply:
i got it.

Reply:
I think `_getHtmlForWebview` itself is enought for (1)